### PR TITLE
[v1alpha3] Streamline build system, improve consistency with other WW OSS projects

### PR DESCRIPTION
**Pull request by twelho**
Streamline the build system of `wksctl` for better consistency with the other Weaveworks OSS projects. This PR includes improvements such as improving the file/folder structure for builds and enabling containerized builds.

Bases on https://github.com/weaveworks/wksctl/pull/214.

### CODE REVIEW

**Comment by bboreham** [comment]
Related: #175

File: Makefile [hunk 1]
```diff
@@ -55,10 +55,10 @@ all: $(UPTODATE_FILES) binaries
...
 
 BINARIES = \
-	cmd/wksctl/wksctl \
-	cmd/mock-authz-server/server \
-	cmd/mock-https-authz-server/server \
-	cmd/controller/controller \
+	bin/wksctl \
+	bin/mock-authz-server \
+	bin/mock-https-authz-server \
+	bin/controller \
```
Comment (bboreham) [inline Makefile:61]: Is there a big benefit from this change?  I think it makes things less consistent with other projects.

Comment (luxas) [reply Makefile:61]: I personally prefer bin for binaries, I think that's much easier to find than individual `cmd/*` locations, also more to type. A quick look yielded:

bin/
 - footloose
 - flagger
 - ignite

/ (top-level)
 - eksctl

cmd/*
 - scope
 - weave-net

So I'm not sure which one is the right way, but coming from the k8s world I'd expect things in `bin`. I quickly checked CAPI core, CAPA, CAPZ and CAPDO, and they all use  `bin/`.

Regardless of what we choose (to be decided), I'd like us to become more consistent in wrt these things.

Comment (bboreham) [reply Makefile:61]: I'd be open to making it consistent with CAP* projects.
@richardcase had some other points here, e.g. using Kustomize, and about where the manifests are located.

**Pull request closed without merging**